service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
I try to elaborate a bit more on my thoughts. We have some function that creates resources. In this case we create EC2 instances. When we create resources the interface of the creating function should return an error and an identifier which can be used to fetch information about the created resources using another function which job is only to fetch information. I want to have the responsibility of creating and reading resources be separated. I got the impression that we have a function to create resources which we now modify to additionally fetch information about the created resources in one step. This would mix up the concepts I described. So I would like to prevent this, because today we need information X, tomorrow we need information y, and then the interface will change again. I am not sure the information we are additionally returning here right now is the identifier that can be used to fetch all the necessary information we will need in the future, so I thought this is a design flaw. Think about it like a rest API.